bootstrap/helpers/parsers.php

```
@@ -16,6 +16,132 @@
+ * @throws \Exception If the compose file contains command injection attempts
+ */
+function validateDockerComposeForInjection(string $composeYaml): void
+{
+    try {
+        $parsed = Yaml::parse($composeYaml);
+    } catch (\Exception $e) {
+        throw new \Exception('Invalid YAML format: '.$e->getMessage());
+    }
+
+    if (! isset($parsed['services']) || ! is_array($parsed['services'])) {
+        throw new \Exception('Docker Compose file must contain a "services" section');
```

PHPMD (2.15.0)

Missing class import via use statement (line '36', column '19'). (undefined)

(MissingImport)